tools/file_packager.py

```diff
@@ -459,23 +460,25 @@ def main():  # noqa: C901, PLR0912, PLR0915
         return 1
     elif leading == 'exclude':
       excluded_patterns.append(arg)
+    elif leading == 'preload-size-limit':
+      options.preload_size_limit = int(arg.split('=', 1)[1] if '=' in arg else arg)
     else:
       err('Unknown parameter:', arg)
       return 1
 
-  options.has_preloaded = any(f.mode == 'preload' for f in data_files)
-  options.has_embedded = any(f.mode == 'embed' for f in data_files)
+  has_preloaded = any(f.mode == 'preload' for f in data_files)
+  has_embedded = any(f.mode == 'embed' for f in data_files)
```

Comment:
This change looks unrelated?  Is so perhaps make this a separate PR?

Reply:
@sbc100 Looks like the plan is to deprecate support for combing of these two modes.  I adjusted this PR so the splitting only applies when you are exclusively preloaded